lib/index.js

```diff
@@ -5,6 +5,7 @@ const Minimatch = require('minimatch').Minimatch;
 const getConfig = require('./get-config');
 const stripBom = require('strip-bom');
 const chalk = require('chalk');
+const { searchAndExtractHbs } = require('extract-tagged-template-literals');
```

Comment:
Hmm, I'm vaguely concerned with a few things in this lib:

* Why does it use regexp's instead of parsing the JS/TS? (using regexp's _seems_ error prone)
* How does the lib support things like `hbs('some template here')`? I made https://github.com/flashios09/extract-tagged-template-literals/issues/2 to discuss.
* I'd like to see an increased bus factor before using it here (I think only @flashios09 has commit + release permissions atm)...
* How do we plan to support updates? We are actively working to move ember-template-lint over to using ember-template-recast (see #305 and #873) so that we can support a `--fix` concept and I'd like to make sure we have a path to support the same functionality in inline templates.
* I would have assumed that we want to process each inline template independently, I'm not super tied to that but its just what I thought we'd want
* I wonder if we should be correcting the location info here in ember-template-lint or in the underlying lib. There are a number of rules that I think could fail given the current setup (things like indentation thinking that the first block should be completely outdented, or that your template doesn't have a bunch of trailing newlines, etc)

Reply:
> Why does it use regexp's instead of parsing the JS/TS? (using regexp's seems error prone)

Why we need a **parser** to **just** extract **tagged literal templates** ? 
and what is the parser to use to go from **js/ts** to **hbs** ?
I have tried:
- **javascript/babylon** -> transform **js/ts** to **js/ts**
- **handlebars/ember-template-recast/glimmer** -> transform **hbs** to **hbs**

> How does the lib support things like hbs('some template here')? I made flashios09/extract-tagged-template-literals#2 to discuss.

The goal was to **add ember template linting to tagged literal templates** inside **ts/js** **component** or **test/component**, like this we can write SFC(personally i prefer SFC especially if we have **short template**)
-> so we extract **inline templates**, replace the script lines with `' '` or `' '`(to preserve the indentation or `\n`) then pass **the converted to hbs source** to `Glimmer::preprocess(...)` to get linting errors/warning.

I don't **think/know** that we **can/have to** do the same for **non tagged literal templates** formats, i mean adding support for **ember template lint** !!
How we can do this with **CallExpression** ? 
What about concatenation ? using vars for templates ?
```js
const input = '<input />'; // ember-template-error self-closing-void-elements
const template = hbs('<div>' + input + '</div>' + '<br>'); 
```
what we will extract here ? the error **self-closing-void-elements** will be attached  to line 1 or line 2 ?

> I'd like to see an increased bus factor before using it here (I think only @flashios09 has commit + release permissions atm)...

I will send you an invitation if you want :trollface: 

> How do we plan to support updates? We are actively working to move ember-template-lint over to using ember-template-recast (see #305 and #873) so that we can support a --fix concept and I'd like to make sure we have a path to support the same functionality in inline templates.

The extracted inline templates(just hbs source) will have the same locations(line, column), the indentations will be preserved(same as written in the original file), so i think it can be parsed with **glimmer** then if we have some errors that we can fix, e.g. from `<input />` to `<input >`, it can be done :)
Otherwise, what is the input format that we have to pass ?

> I would have assumed that we want to process each inline template independently, I'm not super tied to that but its just what I thought we'd want

Why we want to process each inline template **independently** ?

> I wonder if we should be correcting the location info here in ember-template-lint or in the underlying lib.

What location should be corrected ?

> There are a number of rules that I think could fail given the current setup (things like indentation thinking that the first block should be completely outdented, or that your template doesn't have a bunch of trailing newlines, etc)

We can skip/ignore some rules for inline templates like `eol-last` but for the indentation, i don't think that we can touch/remove/outdent it, how to deal with something like this:
```js
const inputTemplate = hbs`<input value={{@value}} />`; // error self-closing-void-elements
const buttonTemplate = hbs`
    <button type={{this.type}}>
         {{yield}}
    </button>
`;
```
will be converted to:
```hbs
                                                <input value={{@value}} />

    <button type={{this.type}}>
         {{yield}}
    </button>
```
If we will force a **de-indentation** or remove the **left whitespaces**, the lint error will not have the correct location(line and column number) ! 

For **trailing spaces**, i have already fix this in `extract-tagged-literal-templates`.
The empty lines at the end(after the last template), will not be added(appended) to the extracted hbs source.

Reply:
> BTW, is it possible to use more than one template for a component/route file ?

No, not really. It is however very common to have many many inline templates in test files.